ChronoVisor/src/ChronicleMetaDirectory.cpp

```diff
@@ -175,20 +196,24 @@ uint64_t ChronicleMetaDirectory::playback_event(uint64_t sid) {
     return 0;
 }
 
-std::string ChronicleMetaDirectory::get_chronicle_attr(std::string &name,
-                                                       const std::string& key) {
+int ChronicleMetaDirectory::get_chronicle_attr(std::string& name, const std::string& key, std::string& value) {
     LOGD("getting attributes key=%s from Chronicle name=%s", key.c_str(), name.c_str());
     uint64_t cid = CityHash64(name.c_str(), name.size());
     Chronicle *pChronicle = chronicleMap_->at(cid);
-    return pChronicle->getPropertyList().at(key);
+    value = pChronicle->getPropertyList().at(key);
```

Comment:
You can add a check for 'chronicle exists' here. Return SUCCESS only if chronicle exists.

Reply:
done